### ISSUE TITLE
bug in chooseColumns followed by View Record?

### ISSUE THREAD

**Issue by bjd-pfq** [free-jqgrid/jqGrid#293]
I am currently experiencing this behaviour: I have a grid with all columns displayed and visible ('standard' situation).
The grid has a columnChooser button.
Now I choose some columns to be hidden and not displayed and click 'OK'. The columns are hidden in the grid. Fine.
Then I select a row and click View Record in the navGridPager (i.e. Form Viewing).
In this form the fields that should be hidden _instead are displayed_.
Then in that form I click the 'Forward' pager button.
The next record properly has the hidden fields hidden!
Then I click the 'Backward' pager button.
Now the previous record is shown again, and this time the columns that should be hidden are correctly hidden.

So in short: the 'hidden'-state seems to be ignored on the _first display of a row_ in the View Form.

P.S.
jqGrid 4.13.6-pre

**Comment by OlegKi** [comment]
Could you provide the demo, which reproduces the problem. I tried [this one](http://www.ok-soft-gmbh.com/jqGrid/OK/viewHiddenColumns.htm), but everything seems to work OK.

**Comment by bjd-pfq** [comment]
I think it may be related to having the record in the View form split up in two columns.  I am also seeing this behaviour: when I have 'disabled' a field with the columnChooser, the field in the View form that is also on that 'level' (but in the other 'column in the View form) is also (inadvertently) invisible.

Example, View form, with two 'columns', with all fields in the grid visible:
aaa ddd
bbb eee
ccc fff

Now with columnChooser disable field 'eee' in the grid.
Now in the View form bbb will also be disabled:
aaa ddd
ccc fff

I'll see what else I can find out tomorrow.

**Comment by OlegKi** [comment]
Sorry, but I can't follow you. You write about some 'level' in the View form and  'disabled' fields in `columnChooser`. The method `columnChooser` can be used to **hide** or **show** some columns or to **reorder** visible columns, but one can't "disable" the columns in some ways. I don't understand the description of the example too.

Please provide **your demo** and describe the exact test case, which can be used to reproduce the problem, which you report. I suppose that there is a problem in combination some specific settings of jqGrid options or properties of `colModel`, but I can't guess how one can reproduce the problem.

**Comment by OlegKi** [comment]
You can use the demo https://jsfiddle.net/OlegKi/0f8obdzc/1/ as the basis and to modify it so that one could reproduce the problem, which you try to report.

**Comment by bjd-pfq** [comment]
OK, I just played with your fiddle. When I add 

formoptions: { rowpos: 1, colpos: 1 },
formoptions: { rowpos: 2, colpos: 1 },
formoptions: { rowpos: 3, colpos: 1 },
formoptions: { rowpos: 4, colpos: 1 },
formoptions: { rowpos: 5, colpos: 1 },
formoptions: { rowpos: 1, colpos: 2 },
formoptions: { rowpos: 2, colpos: 2 },
formoptions: { rowpos: 3, colpos: 2 },
formoptions: { rowpos: 4, colpos: 2 }

to create two 'columns' in the View Form (one of 5 elements and one of 4 elements) to the end of each element in colModel, I am seeing the same behaviour.
Try it!

Your View form now has two columns.
Now in columnChooser hide the column 'Shipped via' en press OK.
Now again open View Form.
You'll see all fields displayed, including 'Shipped via'(!).

Then without closing the View Form, page to the next record with the pager button on the lower left.
Now you won'tl see the 'Shipped via' field anymore, it is correct;y hidden. But notice how the 'Date' field is also hidden(!!).  It would have been on the same 'level', on the same row as the 'Shipped via' field.

**Comment by OlegKi** [comment]
Could you just modify the JSFiddle demo, click on "Save" and to send me the new URL of the modified demo?

**Comment by bjd-pfq** [comment]
Is it the same link?

https://jsfiddle.net/OlegKi/0f8obdzc/1/

I just registered with Fiddle, since I wasn't using it until now.
I saved it and ran it, but the URL is still like you sent me.
Let me know if you can see my changes.

I changed opne thing: earlier I didn't notice that had no
column name, so I skipped that from the View Form. So it now
has 4x4.  Now when you make 'Shipped via' hidden, the other
field that is hidden is 'Amount'.

I hope you can see it. If I needed to send another URL, let me know.


Oleg Kiriljuk wrote on 02/12/2017 11:16 PM:

**Comment by bjd-pfq** [comment]
I think it's this link:
https://jsfiddle.net/bjd_pfq/0f8obdzc/4/


Oleg Kiriljuk wrote on 02/12/2017 11:16 PM:

**Comment by OlegKi** [comment]
Thank you for the demo. I fixed the problem in the latest code of free jqGrid. Not the hidden element will be not displayed. The row with the View-data will be hidden only if all the columns are not vissible. Your demo https://jsfiddle.net/bjd_pfq/0f8obdzc/4/ loads free jqGrid directly from GitHub. Thus you can see the changes directly on the demo. Almost the same demo https://jsfiddle.net/OlegKi/0f8obdzc/5/ uses `hidden: true` property in the `ship_via` column and one can show the View dialog without usage the column chooser before.

If you uses local sources of free jqGrid, you should update there with the latest code from GitHub.

Please test the latest free jqGrid and close the issue, if the problem is solved now.

**Comment by bjd-pfq** [comment]
Hi Oleg,
Yes the issue is fixed. Thank you.   I'll see if I can somehow dynamically modify the formoptions property, to make the View form as compact as possible.

As an aside, I didn't know that setting a hidden property has the same effect as doing it in ColumnChooser, but that looks too static for me; right now I am calling `$("#jqGrid").jqGrid().hideCol( col );` 
in the loadComplete() method, which hides the col in the main grid and in the View form, which is what I want, besides later on dynamically hiding and showing columns with columnChooser.

Anyway, bug fixed -- thanks again!

**Comment by OlegKi** [comment]
You are welcome! Then I'm closing the issue. The `columnChooser` has no side effects. It was a bug in View and Edit form, which worked incorrectly in case of usage `formoptions` if not all data from the row (`rowpos`) were hidden at the same time. Thus now one can hide the column in any way and the View/Edit form should display correctly the data.
